src/nnet3/nnet-training.cc

```diff
@@ -89,14 +108,17 @@ void NnetTrainer::ProcessOutputs(const NnetExample &eg,
       ComputeObjectiveFunction(io.features, obj_type, io.name,
                                supply_deriv, computer,
                                &tot_weight, &tot_objf);
-      objf_info_[io.name].UpdateStats(io.name, config_.print_interval,
-                                      num_minibatches_processed_++,
+      objf_info_[io.name + suffix].UpdateStats(io.name + suffix,
+                                      config_.print_interval,
+                                      (is_adversarial_step ?
+                                      num_minibatches_processed_ :
```

Comment:
no need to do the "?" expression.  Just call with num_minibatches_processed_, and then afterwards, do:
`if (!is_adversarial_step)
    num_minibatches_processed_++;`